app/models/endpoint.rb

```diff
@@ -11,4 +11,24 @@ class Endpoint < ApplicationRecord
   validates :endpoint_node, presence: true
   validates :storage_location, presence: true
   validates :recovery_cost, presence: true
+
+  # rubocop:disable Metrics/AbcSize
+  # TODO: find a clean way to satisfy this style check, everything that silenced the complaint for me was
+  # less readable to my eyes due to what felt like needless indirection.
+  def self.seed_local_online_endpoints_from_config(preservation_policies = nil)
+    default_endpoint_type_config = Settings.endpoint_types.first[1]
+    endpoint_defaults = default_endpoint_type_config.endpoint_defaults
+    endpoint_type = EndpointType.default_local_online_endpoint_type
+    preservation_policies = [PreservationPolicy.default_preservation_policy]
+    Settings.moab.storage_roots.map do |storage_root_name, storage_root_location|
+      find_or_create_by!(endpoint_name: storage_root_name.to_s) do |endpoint|
+        endpoint.endpoint_type = endpoint_type
+        endpoint.endpoint_node = endpoint_defaults.endpoint_node
+        endpoint.storage_location = File.join(storage_root_location, Settings.moab.storage_trunk)
+        endpoint.recovery_cost = endpoint_defaults.recovery_cost
+        endpoint.preservation_policies = preservation_policies
```

Comment:
ditto here:  left side of line 22 could be put on the left side of line 29 ?

Reply:
ditto, `PreservationPolicy.default_preservation_policy` does a db lookup underneath

Reply:
ditto, `PreservationPolicy.default_preservation_policy` does a db lookup underneath

Reply:
in both cases, you're looking up the single record for the default;   the default shouldn't change very often at all;  i think it could be a memoized class method in the appropriate model, and we could have methods in THIS model `default_preservation_policies`  and `default_endpoint_type` that are also memoized class methods here, for convenience.  Then we have nice brief, readable names.